admin/listOrdersJson/endpoint.js

```diff
@@ -1,5 +1,12 @@
 module.exports = (purchases) => (req, res) => {
-    purchases.list((error, orders) => {
+    const options = req.query;
+    Object.keys(options).forEach(key => {
+        if(canParseToInt(options[key])){
+            options[key] = parseInt(options[key]);
+        }  
+    });
```

Comment:
The generalization here kind of ruins the understanding. You have two simple known keys you want to pass in - just parse that out!

Then you can explicitly create the options object, which is also easier to read (and *safer*! if we blindly pass in items from the query param we *may* open ourselves up to unpredictable attacks in the future. A good security principle is to always limit what outside input is passed into our lower level code from an endpoint.)

```js
const { offset, limit } = req.query;

const options = {
    offset: parseIntOrUndefined(offset),
    limit: parseIntOrUndefined(limit),
};

// where I'll use some more lines to make the parsing to integer more explicit/easily readable:

function parseIntOrUndefined(str) {
    const integer = parseInt(str);
    if(isNaN(integer)) {
        return undefined;
    }
    return integer;
}
```

Reply:
Note how the parsing code replaces all these code bits in your current code - so not only is it simpler to parse and more clear what happens, it is also less code:

```js
    const options = req.query;
    Object.keys(options).forEach(key => {
        if(canParseToInt(options[key])){
            options[key] = parseInt(options[key]);
        }  
    });

    // ...

    const offset = getOffset(options);
    const limit = getLimit(options);

    // ...

function getOffset(options){
    if(options["offset"]){
        return options["offset"];
    }
    return 0;
}

function getLimit(options){
    if(options["limit"]){
        return options["limit"];
    }
    return "ALL";
}
```

A lot of the time the best answer is to do the stupid/simple/naïve thing first, and only generalize/extract functions if it makes sense.